src/foam/nanos/dig/DigWebAgent.java

```diff
@@ -115,6 +118,16 @@ public void execute(X x) {
       logger.debug("predicate", pred.getClass(), pred.toString());
       dao = dao.where(pred);
 
+      SinkParser sinkParser = new SinkParser(cInfo);
+
+      for ( int i = 0; i < fieldList.length; i++ ) {
```

Comment:
My understanding is that the parser can handle the entire comma separated string so you don't need to loop over it here. I.e:
```
SinkParser sinkParser = new SinkParser(cInfo);
StringPStream ps = new StringPStream(fields);
ParserContextImpl x_ = new ParserContextImpl();
ps = (StringPStream) sinkParser.parse(ps, x_);
```